### DISCUSSION TITLE
Any reason to not use jests `expect` as a dependency?

### DISCUSSION THREAD

**Discussion by micha149** [vitest-dev/vitest#1052] [General]
Currently I am migrating some tests from jest to vitest. Some tests still fail because vitest does not reimplement the behavior of jest exactly. As I tried to figure out how to fix these issues, I wondered why vitest does not use the standalone library [`expect`](https://www.npmjs.com/package/expect) which seem to be a standalone package of jests expect function.  I think this could reduce the risk of incompatibilities.

**Comment by ScottAwesome** [comment]
I'm not a maintainer of this library, however my general take on it is the following

Chai allows for more flexibility. Chai rusn in both the browser and node environments, so if for instance vitest expanded to have browser testing (like say, via playwright or puppeteer) as an environment, you don't have to make deep changes to the `expect` API based on environment (with the possible exception of snapshots). Where as Jest expect does not work in the browser without _at least_ shimming APIs, and even then, its support is a spotty. they officially dropped `expect` browser support many versions ago. (I think the last browser compatible version was 21.0.0)

That's my "hot" take on it anyway. Chai is simply got more flexibility.

**Reply by micha149** [reply]
> Chai allows for more flexibility.

A long time ago, in my mocha days, I already used chai and have it also very flexible in mind. But in the meantime I used jest _a lot_ and now feel very comfortable with its assertions. In memories on the good old times I also tried to refactor my tests to chai, but in some cases I couldn't express my jest assertions with chai. I think key feature here are the asymetric matchers ( I think they are called so).

But this is not the topic here. It's totally fine to provide both APIs.

> Chai runs in both the browser and node environments [...] Where as Jest expect does not work in the browser [...] they officially dropped expect browser support many versions ago.

OK, this may be a problem I don't had on my screen. But I wonder if we execute the test framework in the browser these days. I don't have any idea about the implementations of tools like puppeteer, playwright and cypress. But I thought the test framework still runs in node and _just_ remotely control a browser to execute your application code. So the assertion will also be done in node. If that's the case, browser compatibility shouldn't be an issue.
